Library/DiscUtils.Ntfs/BiosParameterBlock.cs

```diff
@@ -126,27 +126,29 @@ public void Dump(TextWriter writer, string linePrefix)
         internal static BiosParameterBlock FromBytes(byte[] bytes, int offset)
         {
             BiosParameterBlock bpb = new BiosParameterBlock();
-            bpb.OemId = EndianUtilities.BytesToString(bytes, offset + 0x03, 8);
             bpb.BytesPerSector = EndianUtilities.ToUInt16LittleEndian(bytes, offset + 0x0B);
+            bpb.TotalSectors16 = EndianUtilities.ToUInt16LittleEndian(bytes, offset + 0x13);
+            bpb.TotalSectors32 = EndianUtilities.ToUInt32LittleEndian(bytes, offset + 0x20);
+            bpb.SignatureByte = bytes[offset + 0x26];
+            bpb.TotalSectors64 = EndianUtilities.ToInt64LittleEndian(bytes, offset + 0x28);
+            bpb.MftCluster = EndianUtilities.ToInt64LittleEndian(bytes, offset + 0x30);
+            bpb.RawMftRecordSize = bytes[offset + 0x40];
             bpb.SectorsPerCluster = bytes[offset + 0x0D];
+            if (!bpb.IsValid(long.MaxValue)) return bpb;
```

Comment:
Is this called at other points in time?

Reply:
This is also checked in [NtfsFileSystem.cs#L76](https://github.com/DiscUtils/DiscUtils/pull/104/files/274731cdc77a31f925ae50f2b16a1f83962ce519#diff-e4f867b61d259998b881101f291acc69R76) with the current filesystem length.